samples/saga/migration/sample.md

```diff
@@ -0,0 +1,119 @@
+ 1. Verify that the messages were handled by "Server.New" endpoint.
+ 1. Notice that "Server" console shows information indicating that the not-found handler has been used
+ 1. Ping the previously created sagas (`ping 2` or `ping 3`) to drain the saga store
+ 1. Verify the messages are handled by the old "Server" endpoint, not the "Server.New"
+ 1. Ping one of the new sagas (e.g. `ping A`) to verify it is handled properly "Server.New"
+ 1. Notice that the timeout message that completes the saga does not go through "Server". The destination as stored in the timeout store is set to "Server.New"
+ 1. Ping another saga (e.g. `ping B`) and stop the solution as soon as `Got a follow-up message.` is shown in the console
+ 1. Run `SELECT [Destination], [SagaId] FROM [nservicebus].[dbo].[NewTimeoutData]` to verify the timeout is stored in the database and the destination is the "Server.New" queue
+ 1. Uncomment the `#define POST_MIGRATION` in `Program.cs` of "Server.New". This changes the input queue of "Server.New" back to the well-known `Samples.SagaMigration.Server` and enables an additional receiver that drains the temporary queue. 
+ 1. Start only the "Server.New" project by right-clicking the project in Solution Explorer and selecting "Debug -> Start new instance"
+ 1. Notice "Server.New" prints `Moving message from Samples.SagaMigration.Server.New@<machine> to Samples.SagaMigration.Server@<machine>` and then `Got timeout. Completing.` which means the timeout has been successfully redirected from the temporary queue.
+
+
+## Code walk-through
+
+This sample contains four projects:
+
+ * Contracts - contains definitions of messages exchanged between the Client and the Server.
+ * Client - initiates a multi-message conversation with the server.
+ * Server - implements a long running process via the Saga feature. Uses [NHibernate-based](/nservicebus/nhibernate) saga persister.
+ * Server.New - implements the same functionality as Server but uses [SQL-based](/nservicebus/sql-persistence) saga persister.
+
+The sample shows how to gradually migrate from one saga persister to another without requiring an off-line migration procedure/script. In this example NHibernate and SQL persisters as source and target respectively but any persister can be used in any role i.e. the same method can be used to migrate e.g. from RavenDB to NHibernate persister.
+
+
+### Message flow
+
+The message flow is designed to demonstrate the correctness of migration logic:
+ * The `StartingMessage` (sent via `start` command) starts the saga and assigns the correlation property.
+ * The `CorrelatedMessage` (sent via `ping` command) contains the correlation property value of an already started saga. Handling of this message results in sending back a `ReplyMessage`
+ * The `ReplyMessage` sent by the saga contains the saga ID header containing the storage ID (not the correlation property) of the saga instance that sent it
```

Comment:
I'm not sure I understand the part about the storage ID? I seem to miss anything related to the storage on the returned correlation property?